File: HW/HW2/Main.java
import java.util.Scanner;

class Main {
    static boolean isBadVersion(int version) {
        return version >= 1;
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.print("Enter number of versions: ");
        int n = in.nextInt();

        int low = 1;
        int high = n;
        int verNum = 0;

        while (low <= high) {
            int mid = low + (high - low) / 2;
            boolean bad = isBadVersion(mid);
            if (bad) {
                verNum = mid;
                high = mid - 1;
            }
            else {
                low = mid + 1;
            }
        }

        System.out.print("Version " + verNum + " is the first bad version");
    }
}

File: HW/HW2/main.py
def isBadVersion(version):
    if version >= 70:
        return True
    return False


n = input("Enter number of versions: ")

low = 1
high = int(n)
verNum = 0
print(low, " ", high)

while low <= high:
    mid = int(low + (high - low) / 2)
    bad = isBadVersion(mid)
    print(mid)
    if bad:
        verNum = mid
        high = mid - 1
    if not bad:
        low = mid + 1

print('Version ', verNum, ' is the first bad version')
